R/CITEseq.R

```diff
@@ -1,18 +1,145 @@
 .cord_blood <- function(ess_list)
 {
-    names(ess_list$experiments) <-
-        gsub("_Counts", "", names(ess_list$experiments))
-    MultiAssayExperiment::MultiAssayExperiment(
-        experiments=(ess_list$experiments)
-    )
+    names(ess_list$experiments) <- gsub("_Counts", "", names(ess_list$experiments))
+    mae <- MultiAssayExperiment::MultiAssayExperiment(experiments=(ess_list$experiments))
+    coldat <- sampleMap(mae)[,-c(1:2), drop=FALSE]
+    rownames(coldat) <- coldat[,1]
+    colnames(coldat) <- c("sampleID")
+    return(mae)
 }
 
 
+#' @importFrom Matrix Matrix
+.peripheral_blood <- function(ess_list)
+{
+    .combMatrixForAssay <- function(explist, dimslist, 
+                                    assayId=c("scADT", "scHTO", "scRNA"))
+    {
+        match.arg(assayId)
+        stopifnot( (length(grep(assayId, names(explist)))!=0), 
```

Comment:
do you expect `explist` to not have one of `assayId`?

Reply:
Removing it.